tests/unit/development/test_base_component.py

```diff
@@ -139,128 +137,126 @@ def test_set_item_with_nested_children_with_mixed_strings_and_without_lists(self
                 children='new string'
             )
             c[key] = new_component
-            self.assertEqual(c[new_id], new_component)
+            assert c[new_id] == new_component
 
     def test_del_item_with_nested_children_with_mixed_strings_and_without_lists(self):  # noqa: E501
         c = nested_tree()[0]
         keys = reversed([k for k in c])
         for key in keys:
             c[key]
             del c[key]
-            with self.assertRaises(KeyError):
+            with pytest.raises(KeyError):
                 c[key]
 
     def test_traverse_with_nested_children_with_mixed_strings_and_without_lists(self):  # noqa: E501
```

Comment:
another support point to remove the class and self, you can get rid of the `noqa: E501` too